src/Components/Users/Users.tsx

```diff
@@ -19,15 +31,159 @@ export default function Users() {
     })
       .then((response) => {
         setUserList(response?.data?.data);
-        // setCurrentPage(pageNo);
       })
       .catch((error) => {
         console.log(error);
       })
-    }
+  }
+  //View Users API
+  const getUsersDetails = (id) => {
+    axios
+      .get(`${baseUrl}/Users/${id}`, {
+        headers: requestHeaders,
+      })
+
+      .then((response) => {
+        setUserDetails(response.data);
+
+      })
+      .catch((error) => {
+        console.log(error);
+      })
+  }
+  // toggle activated employee
+  const activatedEmployee = (itemId: number, data: any) => {
+    axios.put(`${baseUrl}/Users/${itemId}`, data, {
+      headers: requestHeaders,
+    })
+      .then((response) => {
+        console.log(response);
+
+        getAllUsers();
+        getToastValue("success", response?.data?.message || "Updated Sucessfully")
+      })
+      .catch((error) => {
+        console.log(error);
+
+        getToastValue("error", error?.response?.data?.message || "Error in updating the status");
+      })
+  }
+
+
+  // Toggle activation status
+  const toggleActivationStatus = (itemId: number, isActivated: boolean) => {
+    const updatedData = { isActivated: !isActivated };
+    activatedEmployee(itemId, updatedData);
+  };
+
+  //view Model
+  const showViewModel = (id: number) => {
+    setItemId(id);
+    setModelState("view-model")
+    getUsersDetails(id)
+  }
+
+  useEffect(() => {
+    getAllUsers()
+  }, [])
+
+  // Search by name
+  const getNameValue = (input) => {
```

Comment:
This function is invoked with each keystroke, resulting in an API call to the backend. However, this approach poses a performance issue, especially when typing longer text, as it leads to sending numerous network requests—where, in reality, only the last one is necessary.

Consider implementing debouncing to address this concern. Debouncing ensures that only one network request is sent after the user completes typing. To understand and implement debouncing, you may find it helpful to search for resources on the topic.

For a more visual explanation of this issue and a potential solution, you can watch this :
https://youtu.be/ZTdz-aBJkUE